### PR TITLE
fix(ord-connect): remove extra horizontal seperators on mobile wallet selection

### PR DESCRIPTION
**Pull request by danadi7**
#### What this PR does / why we need it:
- Fixes extra horizontal separators appearing on mobile

Screenshot 2024-03-07 at 12 46 01

#### Developer Checklist:

- [ ] Read your code changes at least once
- [ ] No console errors on web
- [ ] Your UI implementation visually matched the rendered design\*
- [ ] Unit tests\*
- [ ] Added e2e tests\*

### CODE REVIEW

File: packages/ord-connect/src/components/SelectWalletModal/index.tsx [hunk 1]
```diff
@@ -414,20 +414,22 @@ export function SelectWalletModal({
...
-                          onConnect={onConnectMagicEdenWallet}
-                          icon={MagicEdenWalletIcon}
-                          setErrorMessage={setErrorMessage}
-                          isDisabled={isMobile}
-                          isMobileDevice={isMobile}
-                          renderAvatar={renderAvatar}
-                        />
+                        <>
+                          <hr className="horizontal-separator" />
+                          <WalletButton
```
Comment (kyleleow) [inline packages/ord-connect/src/components/SelectWalletModal/index.tsx:420]: I saw we have multiple `!isMobile` conditions here, can we combine them into one?

Comment (danadi7) [reply packages/ord-connect/src/components/SelectWalletModal/index.tsx:420]: Okay, combined them into one. Originally I was thinking of keeping it separate per wallet but this works too.

File: packages/ord-connect/src/components/SelectWalletModal/index.tsx [hunk 2]
```diff
@@ -414,31 +414,31 @@ export function SelectWalletModal({
                         isMobileDevice={isMobile}
                         renderAvatar={renderAvatar}
                       />
-                      <hr className="horizontal-separator" />
                       {!isMobile && (
```
Comment (kyleleow) [inline packages/ord-connect/src/components/SelectWalletModal/index.tsx:417]: currently `<WalletButton/>` is repeated for 4 times here, we can create a constant in which it is an array of object that consists of whether or not this wallet is available on mobile, e.g.: [{name: xverse, availableOnMobile: true}, {name: unisat, ...}, ... ] 

Then in the rendering logic we can make sure of `.map`

Comment (Nanosync) [reply packages/ord-connect/src/components/SelectWalletModal/index.tsx:417]: I think `<hr />` can also be conditionally rendered based on the number of wallets that will be displayed

Comment (danadi7) [reply packages/ord-connect/src/components/SelectWalletModal/index.tsx:417]: Let's do the refactor another time, we'll remove the extra horizontal separators first

**Review by Nanosync — APPROVED** [review]

**Pull request merged**